src/azul/indexer/document.py

```diff
@@ -340,6 +345,29 @@ def from_index(self, value: Number) -> Optional[bool]:
 
 null_bool: NullableBool = NullableBool()
 
+
+class NullableDateTime(FieldType[Optional[datetime], str]):
+    null_date = '0001-01-01T00:00:00.000000Z'
+    es_type = 'date'
+    format_to_string = '%Y-%m-%dT%H:%M:%S.%fZ'
+    format_from_string = '%Y-%m-%dT%H:%M:%S.%f%z'
```

Comment:
Doesn't the duplication bother you? It's not just that, it's also that if someone changes one and forgets to change the other, things will break in horrible ways.

Reply:
It's probably better to use just one format, assert `"+0000"` at the end of the result and replace with `"Z"`.